File: src/面试题02_02.kth-node-from-end-of-list-lcci.1000007/solution.ts
class ListNode {
    val: number
    next: ListNode | null
    constructor (val?: number, next?: ListNode | null) {
        this.val = (val === undefined ? 0 : val);
        this.next = (next === undefined ? null : next);
    }
}

function kthToLast (head: ListNode | null, k: number): number {
    let fast:ListNode|null = head as ListNode;
    while (k--) {
        fast = (fast as ListNode).next;
    }
    let slow:ListNode|null = head;
    while (fast !== null) {
        fast = fast.next;
        slow = (slow as ListNode).next;
    }
    return (slow as ListNode).val;
}

File: src/面试题02_02.kth-node-from-end-of-list-lcci.1000007/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int kthToLast(ListNode head, int k) {
        ListNode fast = head;
        ListNode slow = head;
        while (k>0){
            k--;
            fast = fast.next;
        }
        while (fast != null){
            fast = fast.next;
            slow = slow.next;
        }
        return slow.val;
    }
}